cloud-aws/src/main/resources/templates/aws-cf-stack.ftl

```diff
@@ -533,6 +533,28 @@
         </#if>
         "ThroughputMode": "${efsFileSystem.throughputMode}"
       }
+    },
+    "MountTarget": {
+      "Type": "AWS::EFS::MountTarget",
+      "Properties": {
+        "FileSystemId": { "Ref": "ElasticFileSystem" },
+        <#if !existingSubnet>
+            "SubnetId": { "Ref": "PublicSubnet" },
+        <#else>
+            "SubnetId": { "Ref": "SubnetId" },
+        </#if>
+        "SecurityGroups" : [
```

Comment:
Putting back my earlier comment that disappeared due to code changes:

According to the [respective AWS CF doc page](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-mounttarget.html), `SecurityGroups` may contain

> Up to five VPC security group IDs

so this limit (if it is a hard one, which would be worth double-checking) may be easily exceeded when having many instancegroups and multiple SGs per instancegroup. I fear it will not be trivial to come up with a perfect solution here.

Reply:
I have tries on AWS console for two scenarios, the observations are
1) When adding security groups ingress_SG as ingress of another security group target_SG (this is the original implementation to create a target_SG for the mount target, then adding the ingress_SG of the EC2 instances as ingress of the target_SG), there is hard limit of only 5 ingress_SG can be added. 
![Security Group ingress does not allow more than 5 SG](https://user-images.githubusercontent.com/26745892/107394451-8656cd00-6ac1-11eb-95cd-6e1c0d1eb6fa.png)
![Finds the same Security Group when ingress has less than 5 Security Groups](https://user-images.githubusercontent.com/26745892/107394529-9c648d80-6ac1-11eb-8833-032489136783.png)
2) When adding security groups ingress_SG to the mount target of EFS directly (this is the update implementation based on Richard suggestion. We don't create a target_SG for the mount target. Instead, we use the ingress_SG of the EC2 instances directly), there is also a hard limit of only 5 ingress_SG can be added. 
![EFS mount target does not allow more than 5 SG](https://user-images.githubusercontent.com/26745892/107394874-ebaabe00-6ac1-11eb-9048-5eb990e0488e.png)

My conclusion is that both approaches have hard-limit of 5 SG can be used for access control for EFS. I can run a local deployment and verify using more than 5 ingress_SG will fail the EFS deployment. 

This indicates when it is possible for customers to specify more than 5 SGs and we want to guarantee the permission works, we need to look for other approaches. One approach I have identified is to use access point instead of mount group. I am experimenting this approach right now.